app/src/main/java/org/mozilla/fenix/utils/Settings.kt

```diff
@@ -135,7 +136,7 @@ class Settings private constructor(
 
     val shouldShowTrackingProtectionOnboarding: Boolean
         get() = trackingProtectionOnboardingCount < trackingProtectionOnboardingMaximumCount &&
-                !trackingProtectionOnboardingShownThisSession
+                !trackingProtectionOnboardingShownThisSession && !overrideTPPopupsForPerformanceTest
```

Comment:
If feel like this would be clearer if we separated these statements with OR: at a high-level, either we're overriding (so don't show onboarding) OR we're checking the conditions for showing this session.